test/generation/test_gen.py

```diff
@@ -69,7 +69,7 @@ def _setUp(self, double=False, expand=False):
 class TestGenCandidates(TestBaseCandidateGeneration):
     def test_gen_candidates(self, gen_candidates=gen_candidates_scipy, options=None):
         options = options or {}
-        options = {**options, "maxiter": 5}
+        options["maxiter"] = options.get("maxiter", 5)
```

Comment:
This is problematic since if options is passed as a dict, this method will now modify the `options` dict in-place. We generally should avoid these kinds of side effects to function arguments (unless we specifically want this to happen).